bin/all_sky_search/pycbc_prepare_xml_for_gracedb

```diff
@@ -0,0 +1,226 @@
+
+sngl_ids = []
+for coinc_map in coinc_event_map_table:
+    if coinc_map.coinc_event_id == event.coinc_event_id:
+        coinc_event_map_table_curr.append(coinc_map)
+        sngl_ids.append(coinc_map.event_id)
+
+# Get the SNR timeseries and PSDs at the time of this event
+# these may not have contributed a trigger
+
+# Check that the SNR timeseries correspond to the event
+# - allow 0.5s either side of the time
+snr_ts = {}
+for ifo, t in snr_timeseries.keys():
+    if not abs(t - time) < 0.5:
+        raise ValueError("SNR timeseries for IFO %s does not look like it "
+            "corresponds to this event, event time %.3f, SNR timeseries is "
+            "around time %.3f" % (ifo, time, t))
+    # Convert the SNR dict to be keyed on IFO-only:
+    snr_ts[ifo] = snr_timeseries[(ifo, t)]
+
+# IFOs from SNR timeseries:
+psds_event = {}
+psddict = {}
+for ifo in snr_ts.keys():
+    psd = psds[ifo]
+
+    psd = psd[psd.find(time)].psd
+    # resample the psd to new spacing
+    psd_fs = FrequencySeries(psd, delta_f=psd.attrs["delta_f"],
+                             dtype=np.float64)
+    psd_fs = interpolate(psd_fs, args.delta_f)
+
+    psds_event[ifo] = psd
+    psddict[ifo] = psd_fs
+
+lal_psddict = {}
+sample_freqs = {}
+
+for sngl in sngl_inspiral_table:
+    if sngl.event_id not in sngl_ids:
+        continue
+    sngl_inspiral_table_curr.append(sngl)
+    sngl.channel = args.channel_name[sngl.ifo]
+
+    # Two versions of the PSD dictionary have useful info here
+    psd = psds_event[sngl.ifo]
+    psd_fs = psddict[sngl.ifo]
+
+    flow = psd.file.attrs['low_frequency_cutoff']
+    kmin = int(flow / args.delta_f)
+
+    fseries = lal.CreateREAL8FrequencySeries(
+        "psd",
+        lal.LIGOTimeGPS(int(psd.attrs["epoch"])),
+        kmin * args.delta_f,
+        args.delta_f,
+        lal.StrainUnit**2 / lal.HertzUnit,
+        len(psd_fs) - kmin)
+    fseries.data.data = psd_fs[kmin:] / np.square(pycbc.DYN_RANGE_FAC)
+    lal_psddict[sngl.ifo] = fseries
+
+    snr_series_to_xml(snr_ts[sngl.ifo], xmldoc, sngl.event_id)
+
+xmldoc.childNodes[-1].appendChild(coinc_event_table_curr)
+xmldoc.childNodes[-1].appendChild(coinc_inspiral_table_curr)
+xmldoc.childNodes[-1].appendChild(coinc_event_map_table_curr)
+xmldoc.childNodes[-1].appendChild(sngl_inspiral_table_curr)
+lal.series.make_psd_xmldoc(lal_psddict, xmldoc.childNodes[-1])
+
+ligolw_utils.write_filename(xmldoc, args.output_file)
+logging.info("Saved XML file %s", args.output_file)
+
+if args.psd_plot:
+    generate_asd_plot(psddict, args.psd_plot)
+    logging.info("Saved ASD plot %s", args.psd_plot)
+
+triggers = {sngl.ifo: (sngl.end_time + sngl.end_time_ns * 1e-9, sngl.snr)
+            for sngl in sngl_inspiral_table_curr}
+base_time = int(np.floor(time))
+if args.snr_timeseries_plot:
+    generate_snr_plot(
+        snr_ts,
+        args.snr_timeseries_plot,
+        triggers,
+        base_time
+    )
+    logging.info("Saved SNR plot %s", args.snr_timeseries_plot)
```

Comment:
Turn the "Saved…" messages into "Saving…" messages just before the corresponding operation (in case of an error we want to know what the code was doing, and we would not see the "Saved" message).

Also, add the usual "Done" at the end.